internal/controller/soperatorchecks/k8s_nodes_controller.go

```diff
@@ -249,6 +267,56 @@ func (c *K8SNodesController) processRebootCondition(ctx context.Context, k8sNode
 	)
 }
 
+func (c *K8SNodesController) processNotReadyCondition(ctx context.Context, k8sNode *corev1.Node) error {
+	logger := log.FromContext(ctx).WithName("K8SNodesController.processNotReadyCondition")
+	logger.Info("processing NotReady condition")
+
+	var readyCondition corev1.NodeCondition
+	for _, cond := range k8sNode.Status.Conditions {
+		if cond.Type == corev1.NodeReady {
+			readyCondition = cond
+			break
+		}
+	}
+
+	if readyCondition.Status == corev1.ConditionTrue || readyCondition.Type == "" {
+		logger.Info("node is ready or condition not found, no action needed")
+		return nil
+	}
+
+	notReadyDuration := time.Since(readyCondition.LastTransitionTime.Time)
+	if notReadyDuration < c.NotReadyTimeout {
+		return nil
+	}
+
+	logger.Info("node is NotReady for more than configured timeout",
+		"duration", notReadyDuration, "timeout", c.NotReadyTimeout)
+
+	logger.Info("deleting k8s node due to NotReady status")
+	return c.deleteK8SNode(ctx, k8sNode)
+}
+
+func (c *K8SNodesController) shouldRequeueForNotReady(k8sNode *corev1.Node) time.Duration {
```

Comment:
`should*` is more about something that returns `true` or `false`
maybe?
```suggestion
func (c *K8SNodesController) requeueDurationForNotReady(k8sNode *corev1.Node) time.Duration {
```